### PR TITLE
[Security] clarified the encoding with memory providers

### PR DESCRIPTION
**Pull request by HeahDude**
Attempt to fix symfony/symfony#35621.

### CODE REVIEW

File: security.rst [hunk 1]
```diff
@@ -449,13 +449,18 @@ To fix this, add an ``encoders`` key:
...
 or :doc:`some other source </security/custom_provider>`, you'll
-use your own custom User class. But when you use the "in memory" provider type,
+use your own custom User class. But when you use the ``memory`` provider type,
 it gives you a :class:`Symfony\\Component\\Security\\Core\\User\\User` object.
 
 Whatever your User class is, you need to tell Symfony what algorithm was
 used to encode the passwords. In this case, the passwords are just plaintext,
 but in a second, you'll change this to use ``bcrypt``.
 
+.. note::
```
Comment (wouterj) [inline security.rst]: ``.. caution::``?

**Review by wouterj — APPROVED** [review]

File: security.rst (hunk 1 above)
Comment (HeahDude) [reply security.rst]: I hesitated :D, still not sure... but since this lead eventually to a bug/limitation and since you asked, then it surely should.

Comment (HeahDude) [reply security.rst]: Done!

**Comment by OskarStark** [comment]
Thank you Jules.

**Pull request merged**